### PR TITLE
[Process] added a deprecation notice

### PR DESCRIPTION
**Pull request by fabpot**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | yes |
| Tests pass? | yes |
| Fixed tickets | #12703, #12725 |
| License | MIT |
| Doc PR | n/a |

### CODE REVIEW

File: src/Symfony/Component/Process/Process.php [hunk 1]
```diff
@@ -1108,11 +1108,13 @@ public function setStdin($stdin)
...
      * This content will be passed to the underlying process standard input.
      *
-     * @param string|null $input The content
+     * @param mixed $input The content
      *
      * @return self The current Process instance
      *
      * @throws LogicException In case the process is running
+     *
+     * @deprecated Passing an object as an input is deprecated since version 2.5 and will be removed in 3.0.
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Process/Process.php]: This marks the method itself as deprecated (that's how IDE will warn about it)

Comment (fabpot) [inline src/Symfony/Component/Process/Process.php]: fixed

**Comment by xabbuh** [comment]
@fabpot I guess you also need to update [some tests](https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Process/Tests/AbstractProcessTest.php#L219-235) (at least the one using a `Stringifiable` instance as input).

**Comment by fabpot** [comment]
@xabbuh Tests are fixed now.

File: src/Symfony/Component/Process/Process.php (hunk 1 above)
Comment (stof) [inline src/Symfony/Component/Process/Process.php]: this does not look fixed. IDEs will still warn about using a deprecated method (there is no phpdoc to deprecate only a specific signature AFAIK)

**Pull request merged**